src/rmp/src/logic_extractor.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#pragma once
+
+#include <unordered_set>
+#include <vector>
+
+#include "abc_library_factory.h"
+#include "db_sta/dbSta.hh"
+#include "logic_cut.h"
+#include "sta/Graph.hh"
+#include "sta/NetworkClass.hh"
+#include "sta/SearchPred.hh"
+
+namespace rmp {
+
+class SearchPredNonReg2AbcSupport : public sta::SearchPredNonReg2
```

Comment:
Please add a class comment as I can't guess what this is from the name.

Reply:
Done.